site/src/components/form.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Form
+description: The form related components in the Atlas Design System
+template: standard
+---
+
+# Form
+
+Forms contain form control elements such as the following:
+
+- [`.button`](./button.md)
+- [`.checkbox`](./checkbox.md)
+- [`.help`](./help.md)
+- [`.input`](./input.md)
+- [`.label`](./label.md)
+- [`.radio`](./radio.md)
+- [`.select`](./select.md)
+- [`.textarea`](./textarea.md)
+- custom form elements
+
+## Form Field
+
+The `.field` component is a container for form control elements to provide consistency.
+It can be used to hold `.control` containers, which holds the individual control elements.
+
+`.field-label` is used on the label. Add `field-label-sm`/`field-label-lg` makes label a little bigger or smaller than the default.
+
+```html
+<div class="field">
+	<label class="field-label field-label-sm label" for="sample-form-input-sm"> Input </label>
+	<div class="control">
+		<input
+			id="sample-form-input-sm"
+			name="title"
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+		/>
+	</div>
+</div>
+<div class="field">
+	<label class="field-label label" for="sample-form-input"> Input </label>
+	<div class="control">
+		<input
+			id="sample-form-input"
+			name="title"
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+		/>
+	</div>
+</div>
+<div class="field">
+	<label class="field-label field-label-lg label" for="sample-form-input-lg"> Input </label>
+	<div class="control">
+		<input
+			id="sample-form-input-lg"
+			name="title"
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+		/>
+	</div>
+</div>
+<div class="field display-flex">
+	<div class="control">
+		<button class="button button-primary margin-right-xxs">Submit</button>
+	</div>
+	<div class="control">
+		<button class="button button-danger">Cancel</button>
+	</div>
+</div>
+```
+
+### Modifiers
+
+`.form-addons` is added to attach controls together.
+
+```html
+<div class="field field-addons">
+	<div class="control">
+		<div class="select">
+			<select>
+				<option>$</option>
+				<option>€</option>
+			</select>
+		</div>
+	</div>
+	<div class="control">
+		<input class="input" type="text" placeholder="Transfer money" />
+	</div>
+	<div class="control">
+		<a class="button button-primary"> Transfer </a>
+	</div>
+</div>
+```
+
+`.field-horizontal` is used for displaying form label and field side by side.
+`field-body` is used on the control element container.
+
+```html
+<div class="field field-horizontal">
```

Comment:
Do you know if field-horizontal is used in docs? Thinking it would be nice to remove this modifier for a couple reasons:
1. Consistent form layout across the site.
2. In practice, using this modifier to stack horizontal fields would result in alignment issues due to varying label widths.

Reply:
I essentially combined the `.field is-grouped` and `.field is-horizontal` into one. `.field is-grouped` is still being used in Docs, but looks like it's mostly for grouping submit + cancel buttons so it can be replaced by flex atomics. REST API Try It has horizontal label and input field, but it's using columns to bypass the second issue you mentioned above. All to say that yes it can be removed.

Reply:
Removed